src/renderers/dom/client/syntheticEvents/SyntheticEvent.js

```diff
@@ -158,11 +150,22 @@ assign(SyntheticEvent.prototype, {
   destructor: function() {
     var Interface = this.constructor.Interface;
     for (var propName in Interface) {
-      this[propName] = null;
+      if (__DEV__) {
+        Object.defineProperty(this, propName, getPooledWarningPropertyDefinition(propName, Interface[propName]));
+      } else {
+        this[propName] = null;
+      }
+    }
+    if (__DEV__) {
+      var noop = function() {};
```

Comment:
I think we have `emptyFunction` somewhere in the codebase. Can you grep for it? Better than allocating it here.